common/src/main/kotlin/org/valkyrienskies/mod/common/command/VSCommands.kt

```diff
@@ -0,0 +1,166 @@
+        RequiredArgumentBuilder.argument<VSCommandSource, T>(name, type)
+
+    fun registerServerCommands(dispatcher: CommandDispatcher<CommandSourceStack>) {
+        dispatcher as CommandDispatcher<VSCommandSource>
+
+        dispatcher.register(
+            literal("vs")
+                .then(literal("delete").then(argument("ships", ShipArgument.ships()).executes {
+                    try {
+                        val r = ShipArgument.getShips(it, "ships").toList() as List<ServerShip>
+                        vsCore.deleteShips(it.source.shipWorld as ServerShipWorld, r)
+                        it.source.sendVSMessage(TextComponent("Deleted ${r.size} ships!"))
+                        r.size
+                    } catch (e: Exception) {
+                        if (e !is CommandRuntimeException) LOGGER.throwing(e)
+                        throw e
+                    }
+                }))
+
+                // Single ship commands
+                .then(
+                    literal("ship").then(
+                        argument("ship", ShipArgument.ships())
+
+                            // Delete a ship
+                            .then(literal("delete").executes {
+                                try {
+                                    vsCore.deleteShips(
+                                        it.source.shipWorld as ServerShipWorld,
+                                        listOf(ShipArgument.getShip(it, "ship") as ServerShip)
+                                    )
+                                    1
+                                } catch (e: Exception) {
+                                    if (e !is CommandRuntimeException) LOGGER.throwing(e)
+                                    throw e
+                                }
+                            })
+
+                            // Rename a ship
+                            .then(
+                                literal("rename")
+                                    .then(argument("newName", StringArgumentType.string())
+                                        .executes {
+                                            try {
+                                                vsCore.renameShip(
+                                                    ShipArgument.getShip(it, "ship") as ServerShip,
+                                                    StringArgumentType.getString(it, "newName")
+                                                )
+                                                1
+                                            } catch (e: Exception) {
+                                                if (e !is CommandRuntimeException) LOGGER.throwing(e)
+                                                throw e
+                                            }
+                                        })
+                            )
+
+                            // Scale a ship
+                            .then(
+                                literal("scale")
```

Comment:
Since this doesn't work maybe comment it out, so that ppl aren't like 'why isn't this working!!!'